File: Benchmarks/OptorComparison.cs
using System;
using System.Text.Patterns;
using System.Text.RegularExpressions;
using BenchmarkDotNet.Attributes;

namespace Benchmarks {
	[ClrJob, CoreJob, CoreRtJob]
	public class OptorComparison {

		Pattern Optor = -(Pattern)"Hello";

		Regex Regex = new Regex("^(Hello)?");

		[Params("Hello", "World")]
		public String Source { get; set; }

		[Benchmark]
		public Result OptorConsume() => Optor.Consume(Source);

		[Benchmark(Baseline = true)]
		public Match RegexMatch() => Regex.Match(Source);

	}
}

File: Benchmarks.FSharp/OptorComparison.fs
namespace Benchmarks

open System.Text.Patterns
open FParsec
open BenchmarkDotNet.Attributes
open BenchmarkDotNet.Diagnosers

[<ClrJob>]
[<MemoryDiagnoser>]
type OptorComparison() =
    let pattern = ~~"Hello"
    let parsec = attempt (opt (pstringCI "Hello"))

    [<Params("Hello", "World")>]
    member val Source = "" with get, set

    [<Benchmark>]
    member this.PatternConsume() = pattern.Consume(this.Source)

    [<Benchmark>]
    member this.ParsecRun() = run parsec this.Source
